tests/unit/test_rayserve_handler.py

```diff
@@ -0,0 +1,149 @@
+# How to run:
+#   env PYTHONPATH=./ pytest tests/unit/test_rayserve_handler.py
+import pandas as pd
+from flask import Flask, jsonify, request
+import multiprocessing
+import requests
+import time
+from unittest.mock import patch
+import json
+import pickle
+import base64
+
+
+from mindsdb_sql import parse_sql
+from .executor_test_base import BaseExecutorTest
+
+
+# The interface of flask implemented belows has been verified to have the same actions as RayServe.
+app = Flask(__name__)
+mock_server_port = 8129
+
+@app.route('/ready', methods=['POST'])
+def ready():
+    rt = {"success": True}
+    return json.dumps(rt)
+
+@app.route('/my_model/train', methods=['POST'])
+def train():
+    data_dic = json.loads(request.data)
+    df_bytes = base64.decodebytes(data_dic["df"].encode())
+    df = pickle.loads(df_bytes)
+    column_set = set(df.columns)
+    assert 3 == len(column_set)
+    assert 'col1' in column_set
+    assert 'col2' in column_set
+    assert 'label' in column_set
+    assert "label" == data_dic["target"]
+    assert isinstance(data_dic["predictor_id"], int)
+    assert "rayserve_predictor" == data_dic["predictor_name"]
+    rt = {"success": True}
+    return json.dumps(rt)
+
+
+@app.route('/my_model/predict', methods=['POST'])
+def predict():
+    data_dic = json.loads(request.data)
+    df_bytes = base64.decodebytes(data_dic["df"].encode())
+    df = pickle.loads(df_bytes)
+    column_set = set(df.columns)
+    assert 2 == len(column_set)
+    assert 'col1' in column_set
+    assert 'col2' in column_set
+    assert "label" == data_dic["target"]
+    assert isinstance(data_dic["predictor_id"], int)
+    assert "rayserve_predictor" == data_dic["predictor_name"]
+    rt = {data_dic["target"]: [1] * len(df)}
+    return json.dumps(rt)
+
+
+class TestRayServe(BaseExecutorTest):
+    global mock_server_port
+
+    def setup_class(cls):
+        BaseExecutorTest.setup_class(cls)
+        processor = multiprocessing.Process(target=app.run, kwargs={"debug": True,
+                                                                    "host": "0.0.0.0", "port": mock_server_port})
+        processor.daemon = True
+        processor.start()
+        i = 0
+        while i < 30:
+            try:
+                resp = requests.post(f"http://127.0.0.1:{mock_server_port}/ready")
```

Comment:
May be better to increase the sleep() time and wait just once at the start to avoid sending that many requests? This approach seems a bit more intensive than it should, imho 😛